### DISCUSSION TITLE
From MySql Workbench diagram to CFW migrations plugin

### DISCUSSION THREAD

**Discussion by blastiko** [wheels-dev/wheels#1277] [Show and tell]
Hi guys
Attach to this post you find a Mysql Workbench script that create all the initial migrations files from a MSW diagram.
From MSW diagram, it create one file per table, with all the fields, reference (with referential integrity) and (optionally) the timestamps fields.
It set olso the index and unique flag on fields (if necessary).
This script require the inflect.py library for pluralizes/singularizes tables name (see instructions on head of script).
This code is free/open source and not complete nor fully tested: feel free to modify it, increase it or fix any bugs.
I hope it is useful.

[mwdtcm.py.zip](https://github.com/cfwheels/cfwheels/files/12504242/mwdtcm.py.zip)

**Comment by bpamiri** [comment]
This is great. Will have to give it a run when I get back into the office.-Peter Amiri Sent from my iPhone (auto-typing mispell disclaimer in effect)On Sep 2, 2023, at 8:42 AM, blastiko ***@***.***> wrote:﻿
Hi guys
Attach to this post you find a Mysql Workbench script that create all the initial migrations files from a MSW diagram.
From MSW diagram, it create one file per table, with all the fields, reference (with referential integrity) and (optionally) the timestamps fields.
It set olso the index and unique flag on fields (if necessary).
This script require the inflect.py library for pluralizes/singularizes tables name (see instructions on head of script).
This code is free/open source and not complete nor fully tested: feel free to modify it, increase it or fix any bugs.
I hope it is useful.
mwdtcm.py.zip

**Comment by blastiko** [comment]
New version 0.2.1 of the script.
Fix some bugs.

[mwdtcm.zip](https://github.com/cfwheels/cfwheels/files/12569185/mwdtcm.zip)
